src/modules/link.cpp

```diff
@@ -1,22 +1,285 @@
+  comm.unsupportedPacketVersionAlarm = (comm.public_command_packet.alarm_bits & ALARM_PACKET_VERSION) != 0x00;
+}
+
+// get data from modules to be sent back to ui. this is called just before sequence count update and crc set
+void updateDataPacket()
+{
+  // only set the lower 7 bits of mode value
+  comm.public_data_packet.mode_value = 0x7f & parameters.ventilationMode;
+  if (parameters.startVentilation)
+  {
+    comm.public_data_packet.mode_value |= MODE_START_STOP;
+  }
+  else
+  {
+    comm.public_data_packet.mode_value &= ~MODE_START_STOP;
+  }
+  // could not find 
+  //    respiratory_rate_measured
+  //    tidal_volume_measured
+  //    battery_level
+  //
+  comm.public_data_packet.tidal_volume_set = parameters.volumeRequested;
+  comm.public_data_packet.tidal_volume_measured = sensors.currentVolume;
+  comm.public_data_packet.respiratory_rate_set = parameters.respirationRateRequested;  // same field on control structure
+  comm.public_data_packet.ie_ratio_set = parameters.ieRatioRequested; // comm.ieRatioRequested; 
+#ifdef SERIAL_DEBUG
+  SERIAL_DEBUG.print("ie_ratio_set from commmand packet: ");
+  SERIAL_DEBUG.println(comm.public_command_packet.ie_ratio_set, DEC);
+#endif
+  
+  comm.public_data_packet.control_state = control.state;
+  // should we use the one from parameters or this one
+  comm.public_data_packet.respiratory_rate_set = control.respirationRateRequested; // // should we use the one from parameters or this one
+  
+  comm.public_data_packet.ie_ratio_measured = control.ieRatioMeasured;
+  //comm.public_data_packet.respiratory_rate_measured = control.  // could not find the field for this
+  
+  // readings from sensor module
+  comm.public_data_packet.plateau_value_measurement = sensors.plateauPressure;
+  comm.public_data_packet.pressure_measured = sensors.currentPressure;  
+  comm.public_data_packet.peak_pressure_measured = sensors.peakPressure;
+  comm.public_data_packet.peep_value_measured = sensors.peepPressure;
+  comm.public_data_packet.tidal_volume_measured = sensors.currentVolume;
+  comm.public_data_packet.volume_in_measured = sensors.volumeIn;
+  comm.public_data_packet.volume_out_measured = sensors.volumeOut;  
+  comm.public_data_packet.volume_rate_measured = sensors.volumePerMinute; 
+  comm.public_data_packet.flow_measured = sensors.currentFlow; 
+//#define SET_VALUES // - for testing
+#ifdef SET_VALUES
+  comm.public_data_packet.mode_value = comm.public_command_packet.mode_value; //comm.ventilationMode;
+  comm.public_data_packet.tidal_volume_set = comm.public_command_packet.tidal_volume_set;
+  comm.public_data_packet.respiratory_rate_set = comm.public_command_packet.respiratory_rate_set;
+  comm.public_data_packet.ie_ratio_set = comm.public_command_packet.ie_ratio_set;
+  comm.public_data_packet.tidal_volume_measured = 100;
+  comm.public_data_packet.respiratory_rate_measured = 200;
+  comm.public_data_packet.ie_ratio_set = 300; 
```

Comment:
s/b "ie_ratio_measured = 300"

Reply:
- [ ] I have now removed the simulation mode line where ie_ratio_set is hardcoded to 300. It is now set to the command packet value. I probably did mean to set the measured value in the first place.